src/components/pages/ProblemListPage/ProblemListPage.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect, useState } from 'react';
+import { getProblemList } from '@services/api/problemService';
+import { Problem } from '@type/problem';
+import { ProblemPageContainer, ProblemsList, ProblemItem } from './ProblemListPage.style';
+import SearchBar from './SearchBar';
+
+const ProblemListPage: React.FC = () => {
+  const [problemList, setProblemList] = useState<Problem[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchProblems = async () => {
+      try {
+        const response = await getProblemList();
+        if (response.status.code === 'SUCCESS_GET_PROBLEM_LIST' && response.result) {
+          setProblemList(response.result.items);
+        } else {
+          setError(`Failed to fetch problems. Status code: ${response.status.code}`);
+        }
+      } catch (err) {
+        setError('An error occurred while fetching problems.');
+      } finally {
```

Comment:
P3: 정상 데이터인지 오류가 있는 데이터인지는 try, catch에서 구분할 수 있어서 별도로 response.status.code를 if문으로 검증하지 않아도 될 것 같습니다!

```suggestion
      try {
        const response = await getProblemList();
        setProblemList(response.result?.items);
      } catch (err) {
          setError(`Failed to fetch problems. Status code: ${response.status.code}`);
      } finally {
```

Reply:
이 부분은 중복된 검증이 있다고 생각합니다. 
근데 if문 없이 response.result를 그냥 사용하면 `'response.result'은(는) 'undefined'일 수 있습니다.ts(18048)` 라는 에러가 떠서 if 문을 추가할 수 밖에 없네요.. lint 메시지도 아니고 ts 자체에서 나는 에러 같습니다. 

차리리 service 단에서 getProblemList에 대한 예외 처리를 하고 이 부분에는 if 문만 사용하는 건 어떨까요?
```typescript 
const fetchProblemList = async () => {
      const response = await getProblemList();
      if (response.status.code === 'SUCCESS_GET_PROBLEM_LIST' && response.result) {
        setProblemList(response.result.items);
      } else {
        setError(`Failed to fetch problems. Status code: ${response.status.code}`);
      }
      setLoading(false);
    };
```

Reply:
비어있는 데이터를 에러로 처리할 지에 대해서는 고민해봐야겠지만  
`'response.result'은(는) 'undefined'일 수 있습니다.ts(18048)` 라는 에러는 아래처럼 리팩토링해봐도 좋을 것 같습니다!
```suggestion
      try {
        const response = await getProblemList();
        const items = response?.result?.items ?? [];
        // items가 null 일 경우 별도 조치가 필요하다면 조건 처리 진행
        setProblemList(items);
      } catch (err) {
        setError('An error occurred while fetching problems.');
      } finally {
```